torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -78,8 +80,22 @@ Message getMessageWithProfiling(
     torch::distributed::rpc::Message&& wrappedRpcMessage,
     MessageType msgType,
     torch::autograd::profiler::ProfilerConfig&& profilerConfig) {
+  auto& remoteProfilerManager =
+      torch::distributed::rpc::RemoteProfilerManager::getInstance();
+  auto key = remoteProfilerManager.getCurrentProfilingKey();
+  // generate a globally unique Id
+  auto localId = remoteProfilerManager.getNextLocalId();
+  auto localWorkerId = RpcAgent::getCurrentRpcAgent()->getWorkerInfo().id_;
+  auto globallyUniqueId =
+      torch::distributed::rpc::GloballyUniqueId(localWorkerId, localId);
+  // Save a mapping of ID -> RPC profiling key and unset the current TLS key.
+  remoteProfilerManager.saveRPCKey(globallyUniqueId, std::move(key));
+  remoteProfilerManager.unsetCurrentKey();
```

Comment:
does it make sense to move these into a helper function in `RemoteProfilerManager`? As it does not seem this function needs to know the `key` or `localId`?

Reply:
Are you referring to `unsetCurrentKey` or other functions as well? We could make `unsetCurrentKey` private and have `saveRPCKey` call it, although, I'd like to reduce side-effects as much as possible. It could be the case that later changes depend on the current key being set, and want to call unsetCurrentKey later in processing the RPC call.

Reply:
I was referring to line 85-93, looks like they can be wrapped in one function and does not need to reveal internal `remoteProfilerManager` and `key` here. But this is a nit, feel free to ignore.

Reply:
Oh I see, yeah I think we should do this.

Reply:
Actually, maybe only the ID generation should be a helper function, since I think it's better to have `saveRPCKey` and `unsetRPCKey` explicit here (or at least in a different helper). Since these functions have side effects, it would be better if they remain explicit.